addon/background/window.js

```diff
@@ -55,6 +55,41 @@ export class Window {
     );
 
     browser.convMsgWindow.onSummarizeThread.addListener(async () => {});
+
+    /**
+     * @typedef {"normal"|"success"|"warning"|"error"} Severity
+     */
+    /**
+     * @typedef {object} AddPillMessage
+     * @property {"addPill"} type
+     *  The type of the received message.
+     * @property {number} msgId
+     *   The id of the associated Message from the WebExtension APIs.
+     * @property {string?} icon
+     *   The optional icon of the pill.
```

Comment:
Should probably note that this needs to be a full moz-extension:// url. Would be good if we could add a check for that in the listener as well - we could fallback to the internal icon if it is an incorrect url.

Reply:
Ok, will add a check for it.

Another think I forgot to mention is that we need to decide on the handling of the optional properties.
Currently they must be explicitly set (at least according to this typedef), i.e. the `icon` must be set to `null` and the `tooltip` to an empty array. Don't know if you also want to allow `undefined` for both, i.e. so that a user can just leave them out in the message.
And we may want to make the `severity` optional too.

Reply:
Yeah, I think we can make icon, tooltip and severity optional.

Reply:
Added the check for `moz-extension://`, and made all besides `msgId` and `message` optional.